positions-transport/src/main/java/workshop/positions/PositionsPusher.java

```diff
@@ -0,0 +1,142 @@
+package workshop.positions;
+
+import static workshop.shared.Constants.DATAGRID_HOST;
+import static workshop.shared.Constants.DATAGRID_PORT;
+import static workshop.shared.Constants.POSITIONS_TRANSPORT_URI;
+import static workshop.shared.Constants.TRAIN_POSITIONS_CACHE_NAME;
+import static workshop.shared.Constants.TRAIN_POSITIONS_TOPIC;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.client.hotrod.marshall.ProtoStreamMarshaller;
+import org.infinispan.protostream.FileDescriptorSource;
+import org.infinispan.protostream.SerializationContext;
+
+import io.reactivex.Single;
+import io.vertx.config.ConfigRetrieverOptions;
+import io.vertx.config.ConfigStoreOptions;
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+import io.vertx.kafka.client.consumer.KafkaReadStream;
+import io.vertx.reactivex.CompletableHelper;
+import io.vertx.reactivex.config.ConfigRetriever;
+import io.vertx.reactivex.core.AbstractVerticle;
+import io.vertx.reactivex.ext.web.Router;
+import io.vertx.reactivex.ext.web.RoutingContext;
+import workshop.model.GeoLocBearing;
+import workshop.model.TimedPosition;
+import workshop.model.TrainPosition;
+
+/**
+ * Listens Kafka and inserts data in Infinispan
+ */
+public class PositionsPusher extends AbstractVerticle {
+
+  private static final Logger log = Logger.getLogger(PositionsPusher.class.getName());
+
+  private KafkaReadStream<String, String> consumer;
+  private RemoteCacheManager client;
+  private RemoteCache<String, TrainPosition> cache;
+
+  @Override
+  public void start(Future<Void> future) {
+    Router router = Router.router(vertx);
+    router.get(POSITIONS_TRANSPORT_URI).handler(this::consumeAndPushToCache);
+
+    retrieveConfiguration()
+      .doOnSuccess(json ->
+        consumer = KafkaReadStream.create(vertx.getDelegate(), json.getJsonObject("kafka").getMap(), String.class, String.class))
+      .flatMapCompletable(v ->
+        vertx.createHttpServer()
+          .requestHandler(router::accept)
+          .rxListen(8080)
+          .doOnSuccess(server -> log.info("Positions transport HTTP server started"))
+          .doOnError(t -> log.log(Level.SEVERE, "Positions transport HTTP server failed to start", t))
+          .toCompletable() // Ignore result
+      )
+      .subscribe(CompletableHelper.toObserver(future));
+  }
+
+  private void consumeAndPushToCache(RoutingContext ctx) {
+    vertx.<Infinispan>rxExecuteBlocking(fut -> fut.complete(createClient()))
+      .doOnSuccess(infinispan -> {
+        log.info("Connected to Infinispan");
+        client = infinispan.remoteClient;
+        cache = infinispan.cache;
+        consumer.handler(record -> {
```

Comment:
`consumer` is a `ReadStream`. It can be converted to `Flowable`. Then the code that we had before to avoid pushing too many items at one to the cache can be reused.

Reply:
I'm going to do this in another PR @tsegismont ! I told galder to merge this, so after snowcamp on friday, I will code it and do additional cleanup on duplicated code in another PR ! 💃

Reply:
👍